starlite/middleware/session/base.py

```diff
@@ -276,17 +277,18 @@ async def store_in_message(
         headers = MutableHeaders(scope=message)
         session_id = connection.cookies.get(self.config.key, self.generate_session_id())
 
-        serialised_data = self.serlialize_data(scope_session, scope)
-        await self.set(session_id=session_id, data=serialised_data)
-
         cookie_params = self.config.dict(
             exclude_none=True,
             exclude={"secret", "key"} | set(self.config.__fields__) - set(BaseBackendConfig.__fields__),
         )
 
-        if scope_session:
+        if scope_session is not Empty:
```

Comment:
Yes I removed `Empty`. If the session is cleared it becomes an empty dictionary because the user should be able to refill the session after clearing it and for an empty session (as in empty dict) this block should not run.

Reply:
Hm. I guess you have a point there, however, removing the `Empty` check breaks starlite-sessions. That’s why I re-added it.

Reply:
why would the user refill the same dictionary? Also, the `scope_session` is a value stored in the scope, its not support to be exposed directly to used as such. So `Empty` is correct here IMO.

Reply:
It also makes it easier for the server-side backends to know when to explicity delete a session instead of replacing it with an empty one.

Reply:
@Goldziher What I mean is if the user does `request.session.clear`, the session becomes an empty dictionary for which this if condition should not become True.

Reply:
@infohash You're right about that, this is actually a mistake. Wonder why it's not caught by the tests.. I implemented this correctly for the cookie backend. Will update accordingly!